src/devices/Mpu9250/Mpu6500.cs

```diff
@@ -0,0 +1,899 @@
+        private GyroscopeBandwidth _gyroscopeBandwidth;
+        internal bool _wakeOnMotion;
+
+        /// <summary>
+        /// Default address for MPU9250
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x68;
+
+        /// <summary>
+        /// Second address for MPU9250
+        /// </summary>
+        public const byte SecondI2cAddress = 0x69;
+
+        /// <summary>
+        /// Initialize the MPU6500
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device</param>
+        public Mpu6500(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;
+            Reset();
+            PowerOn();
+            if (!CheckVersion())
+                throw new IOException($"This device does not contain the correct signature 0x71 for a MPU6500");
+            GyroscopeBandwidth = GyroscopeBandwidth.Bandwidth0250Hz;
+            GyroscopeRange = GyroscopeRange.Range0250Dps;
+            AccelerometerBandwidth = AccelerometerBandwidth.Bandwidth1130Hz;
+            AccelerometerRange = AccelerometerRange.Range02G;
+        }
+
+        #region Accelerometer
+
+        /// <summary>
+        /// Accelerometer bias data
+        /// </summary>
+        public Vector3 AccelerometerBias => _accelerometerBias;
+
+        /// <summary>
+        /// Get or set the accelerometer range
+        /// </summary>
+        public AccelerometerRange AccelerometerRange
+        {
+            get => _accelerometerRange;
+
+            set
+            {
+                WriteRegister(Register.ACCEL_CONFIG, (byte)((byte)value << 3));
+                // We will cache the range to avoid i2c access every time this data is requested
+                // This allow as well to make sure the stored data is the same as the one read
+                _accelerometerRange = (AccelerometerRange)(ReadByte(Register.ACCEL_CONFIG) >> 3);
+                if (_accelerometerRange != value)
+                    throw new IOException($"Can set {nameof(AccelerometerRange)}, desired value {value}, stored value {_accelerometerRange}");
+            }
+        }
+
+        /// <summary>
+        /// Get or set the accelerometer bandwidth
+        /// </summary>
+        public AccelerometerBandwidth AccelerometerBandwidth
+        {
+            get => _accelerometerBandwidth;
+
+            set
+            {
+                WriteRegister(Register.ACCEL_CONFIG_2, (byte)value);
+                _accelerometerBandwidth = (AccelerometerBandwidth)ReadByte(Register.ACCEL_CONFIG_2);
+                if (_accelerometerBandwidth != value)
+                    throw new IOException($"Can set {nameof(AccelerometerBandwidth)}, desired value {value}, stored value {_accelerometerBandwidth}");
+            }
+        }
+
+        /// <summary>
+        /// Get the real accelerometer bandwidth. This allows to calculate the real
+        /// degree per second
+        /// </summary>
+        public float AcceleromterConversion
```

Comment:
I'd rename this to something like `AccelerationScale`/`AccelerationRange`

Reply:
I went for Scale for both